arches_lingo/src/arches_lingo/components/header/DarkModeToggle.vue

```diff
@@ -0,0 +1,37 @@
+<script setup lang="ts">
+import { ref } from "vue";
+import { useGettext } from "vue3-gettext";
+
+import ToggleButton from "primevue/togglebutton";
+
+import { DEFAULT_THEME } from "@/arches/themes/default.ts";
+
+const { $gettext } = useGettext();
+
+const isDarkModeEnabled = ref(
+    window.matchMedia &&
+        window.matchMedia("(prefers-color-scheme: dark)").matches,
+);
+
+const toggleDarkMode = () => {
+    const element = document.querySelector("html");
+    // Remove the leading dot from the selector to get class name.
+    element!.classList.toggle(
+        DEFAULT_THEME.theme.options.darkModeSelector.substring(1),
+    );
+    isDarkModeEnabled.value = !isDarkModeEnabled.value;
+};
+</script>
+
+<template>
+    <ToggleButton
```

Comment:
Started with a binary toggle.

I know the linked arches issue for theme switching mentions three themes. I think high contrast mode for accessibility is its own toggle (perhaps not exposed in the UI, perhaps reads from the existing system setting or the browser settings?) and will affect both light and dark modes. We'll probably want to use the primevue utilities for updating themes in that case. I don't think we want to _define new themes_ per se.

Reply:
yeah happy to punt 🏈